nroam.el

```diff
@@ -185,12 +185,18 @@ Make the region inserted by BODY read-only, and marked with
 
 (defun nroam--insert ()
   "Insert nroam sections in the current buffer."
-  (with-buffer-modified-unmodified
-   (save-excursion
-     (goto-char (point-max))
-     (with-nroam-markers
-       (seq-do #'funcall nroam-sections)))
-   (nroam--hide-drawers)))
+  (let ((p (point-max)))
+    (with-buffer-modified-unmodified
+     (save-excursion
+       (goto-char p)
+       (with-nroam-markers
+         (seq-do #'funcall nroam-sections))
+       (when (nroam--sections-inserted-p)
+         (goto-char (+ p 1)) ; maybe to implicit. But loc. of new
```

Comment:
Does this always work? It seems to be relying on an implementation detail: the position where the "linked references" heading gets inserted.

Would something like the following work instead? (untested)

```elisp
(narrow-to-region p (point-max))
(org-set-startup-visibility)
```

Reply:
Another reason why I'm not comfortable with the `(+ p 1)` part: I'm about to add support for unlinked references (and I'm thinking about a section that would pull scheduled todos from org-agenda and insert them in daily notes), which would most probably break with your changes.

Reply:
Yes, It's ugly... But since point is just a number I can't see it failing as it stands today. It would be better with an abstraction though, for sure. Something like a `nroam--goto-unlinked-references` function that would take care of moving the point to that location, or return the point of that location.

I'll battle-test your implementation a bit. I'm sure it's better :)